nimble/host/src/ble_gap.c

```diff
@@ -737,6 +748,39 @@ ble_gap_slave_reset_state(uint8_t instance)
 #endif
 }
 
+/**
+ * Clears advertisement and discovery state.  This function is necessary
+ * when the controller loses its active state (e.g. on stack reset).
+ */
+void
+ble_gap_reset_state(uint16_t conn_handle, int reason)
+{
+#if NIMBLE_BLE_ADVERTISE
+#if MYNEWT_VAL(BLE_EXT_ADV)
+    uint8_t i;
+
+    for (i = 0; i < BLE_ADV_INSTANCES; i++) {
+        ble_gap_slave_reset_state(i);
+        /* Indicate to application that advertising has stopped. */
+        ble_gap_adv_finished(i, reason, conn_handle, 0);
```

Comment:
I don't think we need `conn_handle` here. It is used when advertising is completed because connection has been established and it is not a case here for sure

Reply:
True.